### PR TITLE
chore: add client suffix on debug builds [ACOL-144]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACOL-144" title="ACOL-144" target="_blank">TopicACOL-144</a>  Add BUILD_TYPE and FLAVOR to client model name for dev or private builds
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When testing different app builds it's hard to know which device is prod build to not remove while adding new device 

### Solutions

In private builds add to device name flavor and build type

Preview

### Dependencies

This PR depends on:

- https://github.com/wireapp/kalium/pull/2480

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/authentication/create/code/CreateAccountCodeViewModel.kt [hunk 1]
```diff
@@ -209,7 +210,8 @@ class CreateAccountCodeViewModel @Inject constructor(
         clientScopeProviderFactory.create(userId).clientScope.getOrRegister(
             RegisterClientUseCase.RegisterClientParam(
                 password = password,
-                capabilities = null
+                capabilities = null,
+                modelPostfix = if(BuildConfig.PRIVATE_BUILD) " [${BuildConfig.FLAVOR}_${BuildConfig.BUILD_TYPE}]" else null
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/authentication/create/code/CreateAccountCodeViewModel.kt]: Is PrivateBuild the correct parameter? Won't clients with custom builds also have that?

**Review by mchenani — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3019](https://10.10.124.134/job/AR-build-pipeline/3019/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/authentication/create/code/CreateAccountCodeViewModel.kt (hunk 1 above)
Comment (vitorhugods) [reply app/src/main/kotlin/com/wire/android/ui/authentication/create/code/CreateAccountCodeViewModel.kt]: Nevermind.

It seems we use `PRIVATE_BUILD` just for debug stuff. Only visible to us.

I guess we could name it better 🤔 
Anyway. LGTM!

**Review by vitorhugods — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/authentication/create/code/CreateAccountCodeViewModel.kt (hunk 1 above)
Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/ui/authentication/create/code/CreateAccountCodeViewModel.kt]: Just checked and as default we go with `false`, if not set (like the case of those custom builds)
https://github.com/wireapp/wire-android/blob/0f6bace635f79c3ee6ec416afe15759e97153b40/buildSrc/src/main/kotlin/scripts/variants.gradle.kts#L178-L183

**Review by yamilmedina — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3030](https://10.10.124.134/job/AR-build-pipeline/3030/) **failed**.

**Comment by AndroidBob** [comment]
Build [3031](https://10.10.124.134/job/AR-build-pipeline/3031/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(8851362)(8851362)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(8851362)(8851362)-staging-compat.apk)

**Pull request merged**